plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcMetadata.java

```diff
@@ -237,11 +227,8 @@ public JdbcTableHandle getTableHandle(ConnectorSession session, SchemaTableName
             return Optional.empty();
         }
 
-        // Global aggregation is represented by [[]]
-        verify(!groupingSets.isEmpty(), "No grouping sets provided");
-
-        if (groupingSets.size() > 1 && !jdbcClient.supportsGroupingSets()) {
-            return Optional.empty();
+        if (handle.getLimit().isPresent()) {
+            handle = flushAttributesAsQuery(session, handle);
```

Comment:
I don't get this one :/

Reply:
the conditions here are a bit reordered (needed to be), so the diff is not very smart